src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -223,9 +221,14 @@ private void EmitResource(ResourceSymbol resourceSymbol)
 
             var typeReference = EmitHelpers.GetTypeReference(resourceSymbol);
 
+            if (resourceSymbol.DeclaringResource.Body is IfExpressionSyntax conditionalBody)
+            {
+                this.emitter.EmitProperty("condition", conditionalBody.ConditionExpression);
```

Comment:
Btw, do we have any limitations on what can go into resource conditions in the runtime? Are all functions including stuff like `reference()` or `listKeys()` allowed?

Reply:
Good questions. The documents didn't mention any limitations, but let me do some tests to figure out.

Reply:
My expectation was deploy-time constants only (similar to `name`, `type`, `apiVersion` field requirements) because the engine calculates the deployment graph upfront. We should probably make a note to clarify this in the docs once you've tested this.

Reply:
Confirmed with template deployments that runtime functions are not allowed in a resource condition (also found this [ARM unit test](https://dev.azure.com/msazure/One/_git/AzureUX-Deployments?path=%2Fsrc%2FTemplates.Tests%2FTemplateParameterizationTests.cs&version=GBmaster&line=4939&lineEnd=4939&lineStartColumn=1&lineEndColumn=88&lineStyle=plain&_a=contents) that validates it). I'll update the code to block the use of them, and I think @anthony-c-martin is right that we should update the docs to clarify this.

Reply:
`reference()` and `list*` functions are blocked now.  We still need to block resource and module symbolic name references, which I will implement in a separate PR once Tarun's PR is merged.